src/services/product/product.service.ts

```diff
@@ -96,4 +96,28 @@ export class ProductService {
       throw new NotFoundException(`Product with id ${id} not found`);
     }
   }
+
+  async validateStock(id: string, quantity: number): Promise<boolean> {
+    const product = await this.productRepository.findById(id);
+    if (!product) {
+      throw new NotFoundException(`Product with id ${id} was not found`);
+    }
+
+    let result = false;
+    if (product.stock >= quantity) {
+      result = true;
+    }
+
+    return result;
+  }
+
+  async updateStock(id: string, quantity: number): Promise<void> {
+    const product = await this.productRepository.findById(id);
+    if (!product) {
+      throw new NotFoundException(`Product with id ${id} was not found`);
+    }
+
+    product.stock -= quantity;
```

Comment:
I think you should make some extra checks here, because you could end with negative stock. by instance `product.stock:1` and `quantity:5`